### PR TITLE
Compile NESTML generated code multithreaded

### PR DESCRIPTION
**Pull request by clinssen**

### CODE REVIEW

File: pynestml/codegeneration/nest_builder.py [hunk 1]
```diff
@@ -123,9 +123,11 @@ def build(self) -> None:
                 install_path = os.path.abspath(install_path)
             install_prefix = f"-DCMAKE_INSTALL_PREFIX={install_path}"
 
+        n_cpu = len(os.sched_getaffinity(0))  # compile multithreaded
```
Comment (pnbabu) [inline pynestml/codegeneration/nest_builder.py:126]: This function is unfortunately not present for Mac OS :(
Could we use `os.cpu_count()` instead?

Comment (clinssen) [reply pynestml/codegeneration/nest_builder.py:126]: ``sched_getaffinity()`` has some advantages on *nix, so I'd like to keep it in. Do you get an exception when running this under the Mac, or just a return of ``None``? We would then fall back to ``os.cpu_count()``.

Comment (pnbabu) [reply pynestml/codegeneration/nest_builder.py:126]: I get this exception: `AttributeError: module 'os' has no attribute 'sched_getaffinity'`. Probably, we can add different functions depending on the platform?

**Review by jougs — APPROVED** [review]
Just a minor comment for readability.

**Pull request merged**